src/main/scala/sigmastate/lang/SigmaParser.scala

```diff
@@ -64,18 +64,23 @@ object SigmaParser extends Exprs with Types with Core {
     case ">"  => GT(l, r)
     case "<=" => LE(l, r)
     case "<"  => LT(l, r)
-    case "+"  => StdSigmaBuilder.Plus(l.asValue[SLong.type], r.asValue[SLong.type])
-    case "-"  => StdSigmaBuilder.Minus(l.asValue[SLong.type], r.asValue[SLong.type])
+    case "+"  => builder.Plus(l.asValue[SLong.type], r.asValue[SLong.type])
+    case "-"  => builder.Minus(l.asValue[SLong.type], r.asValue[SLong.type])
     case "|"  => Xor(l.asValue[SByteArray], r.asValue[SByteArray])
     case "++" => MethodCall(l, "++", IndexedSeq(r))
     case "^"  => Exponentiate(l.asValue[SGroupElement.type], r.asValue[SBigInt.type])
     case "*"  => MethodCall(l, "*", IndexedSeq(r))
-    case "/"  => StdSigmaBuilder.Divide(l.asValue[SLong.type], r.asValue[SLong.type])
-    case "%"  => StdSigmaBuilder.Modulo(l.asValue[SLong.type], r.asValue[SLong.type])
+    case "/"  => builder.Divide(l.asValue[SLong.type], r.asValue[SLong.type])
+    case "%"  => builder.Modulo(l.asValue[SLong.type], r.asValue[SLong.type])
     case _ => error(s"Unknown binary operation $opName")
   }
 
-  def apply(str: String): core.Parsed[Value[_ <: SType], Char, String] = (StatCtx.Expr ~ End).parse(str)
+  private var builder: SigmaBuilder = StdSigmaBuilder
+
```

Comment:
@ergomorphic I found this super ugly but it's all I could come up with. Any ideas on how to make eyes to not bleed are very welcome. :)